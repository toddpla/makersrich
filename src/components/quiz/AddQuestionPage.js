import React, { Component } from 'react';

import {connect } from 'react-redux'
import { startAddQuestion } from '../../actions/questions'

class AddQuestionPage extends Component {

  handleAddQuestion = (e) => {
    e.preventDefault()
    const question = {
      question: e.target.question.value,
      answers: [
        e.target.answer1.value,
        e.target.answer2.value,
        e.target.answer3.value,
        e.target.answer4.value
      ],
      correctAnswer: e.target.correctAnswer.value
    }
    console.log('here')
    this.props.startAddQuestion(question)
  }

  render() {
    return (
      <div>
        <form onSubmit={this.handleAddQuestion}>
          <input type='text' id='question' placeholder='question' />
          <input type='text' id='answer1' placeholder='answer 1' />
          <input type='text' id='answer2' placeholder='answer 2' />
          <input type='text' id='answer3' placeholder='answer 3' />
          <input type='text' id='answer4' placeholder='answer 4' />
          <input type='text' id='correctAnswer'  placeholder='correct answer' />
          <input type='submit' value='submit' />
        </form>
      </div>
    );
  }

}

const mapDispatchToProps = (dispatch) => ({
  startAddQuestion: question => dispatch(startAddQuestion(question))
})

export default connect(undefined, mapDispatchToProps)(AddQuestionPage)
