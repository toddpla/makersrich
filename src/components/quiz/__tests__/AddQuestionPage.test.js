import React from 'react';
import { shallow } from 'enzyme'
import { AddQuestionPage } from '../AddQuestionPage'
import event from '../../../test/fixtures/addQuestionEvent'

let wrapper, startAddQuestion

beforeEach(function() {
  startAddQuestion = jest.fn()
  wrapper = shallow(
    <AddQuestionPage startAddQuestion={startAddQuestion} />
  )
});

describe('AddQuestionPage', function() {
  it('renders', function() {
    expect(wrapper).toMatchSnapshot()
  });

  it('calls #startAddQuestion', function() {
    const question = {
      question:'What is 1 + 1?',
      answers: [
        '1',
        '2',
        '3',
        '4'
      ],
      correctAnswer: '2'
    }
    wrapper.find('form').simulate('submit', event)
    expect(startAddQuestion).toHaveBeenLastCalledWith(question)
  });

});
