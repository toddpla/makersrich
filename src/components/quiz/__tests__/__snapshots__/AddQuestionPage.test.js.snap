// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddQuestionPage renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddQuestionPage
    startAddQuestion={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={[Function]}
      >
        <textarea
          id="question"
          placeholder="Information"
          type="text"
        />
        <textarea
          id="question"
          placeholder="Question"
          type="text"
        />
        <input
          id="answer1"
          placeholder="Answer 1"
          type="text"
        />
        <input
          id="answer2"
          placeholder="Anwser 2"
          type="text"
        />
        <input
          id="answer3"
          placeholder="Anwser 3"
          type="text"
        />
        <input
          id="answer4"
          placeholder="Answer 4"
          type="text"
        />
        <input
          id="correctAnswer"
          placeholder="Correct answer"
          type="number"
        />
        <input
          type="submit"
          value="Submits"
        />
      </form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <textarea
            id="question"
            placeholder="Information"
            type="text"
          />,
          <textarea
            id="question"
            placeholder="Question"
            type="text"
          />,
          <input
            id="answer1"
            placeholder="Answer 1"
            type="text"
          />,
          <input
            id="answer2"
            placeholder="Anwser 2"
            type="text"
          />,
          <input
            id="answer3"
            placeholder="Anwser 3"
            type="text"
          />,
          <input
            id="answer4"
            placeholder="Answer 4"
            type="text"
          />,
          <input
            id="correctAnswer"
            placeholder="Correct answer"
            type="number"
          />,
          <input
            type="submit"
            value="Submits"
          />,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "question",
            "placeholder": "Information",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "textarea",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "question",
            "placeholder": "Question",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "textarea",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "answer1",
            "placeholder": "Answer 1",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "answer2",
            "placeholder": "Anwser 2",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "answer3",
            "placeholder": "Anwser 3",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "answer4",
            "placeholder": "Answer 4",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "correctAnswer",
            "placeholder": "Correct answer",
            "type": "number",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "type": "submit",
            "value": "Submits",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={[Function]}
        >
          <textarea
            id="question"
            placeholder="Information"
            type="text"
          />
          <textarea
            id="question"
            placeholder="Question"
            type="text"
          />
          <input
            id="answer1"
            placeholder="Answer 1"
            type="text"
          />
          <input
            id="answer2"
            placeholder="Anwser 2"
            type="text"
          />
          <input
            id="answer3"
            placeholder="Anwser 3"
            type="text"
          />
          <input
            id="answer4"
            placeholder="Answer 4"
            type="text"
          />
          <input
            id="correctAnswer"
            placeholder="Correct answer"
            type="number"
          />
          <input
            type="submit"
            value="Submits"
          />
        </form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <textarea
              id="question"
              placeholder="Information"
              type="text"
            />,
            <textarea
              id="question"
              placeholder="Question"
              type="text"
            />,
            <input
              id="answer1"
              placeholder="Answer 1"
              type="text"
            />,
            <input
              id="answer2"
              placeholder="Anwser 2"
              type="text"
            />,
            <input
              id="answer3"
              placeholder="Anwser 3"
              type="text"
            />,
            <input
              id="answer4"
              placeholder="Answer 4"
              type="text"
            />,
            <input
              id="correctAnswer"
              placeholder="Correct answer"
              type="number"
            />,
            <input
              type="submit"
              value="Submits"
            />,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "question",
              "placeholder": "Information",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "textarea",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "question",
              "placeholder": "Question",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "textarea",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "answer1",
              "placeholder": "Answer 1",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "answer2",
              "placeholder": "Anwser 2",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "answer3",
              "placeholder": "Anwser 3",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "answer4",
              "placeholder": "Answer 4",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "correctAnswer",
              "placeholder": "Correct answer",
              "type": "number",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "type": "submit",
              "value": "Submits",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
